cloud/lxd/lxd_container.py

```diff
@@ -0,0 +1,572 @@
+#
+# [containers]
+# mycontainer ansible_connection=lxd
+#
+- hosts:
+    - mycontainer
+  tasks:
+    - template: src=foo.j2 dest=/etc/bar
+
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: Create a stopped container
+      lxd_container:
+        name: mycontainer
+        state: stopped
+        config:
+          source:
+            type: image
+            mode: pull
+            server: https://images.linuxcontainers.org
+            protocol: lxd
+            alias: "ubuntu/xenial/amd64"
+          profiles: ["default"]
+
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: Restart a container
+      lxd_container:
+        name: mycontainer
+        state: restarted
+        config:
+          source:
+            type: image
+            mode: pull
+            server: https://images.linuxcontainers.org
+            protocol: lxd
+            alias: "ubuntu/xenial/amd64"
+          profiles: ["default"]
+"""
+
+RETURN="""
+lxd_container:
+  description: container information
+  returned: success
+  type: object
+  contains:
+    addresses:
+      description: mapping from the network device name to a list of IPv4 addresses in the container
+      returned: when state is started or restarted
+      type: object
+      sample: {"eth0": ["10.155.92.191"]}
+    old_state:
+      description: the old state of the container
+      returned: when state is started or restarted
+      sample: "stopped"
+    logs:
+      descriptions: the logs of requests and responses
+      returned: when requests are sent
+    actions:
+      description: list of actions performed for the container
+      returned: success
+      type: list
+      sample: ["create", "start"]
+"""
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+# httplib/http.client connection using unix domain socket
+import socket
+try:
+    import httplib
+
+    class UnixHTTPConnection(httplib.HTTPConnection):
+        def __init__(self, path, host='localhost', port=None, strict=None,
+                     timeout=None):
+            httplib.HTTPConnection.__init__(self, host, port=port, strict=strict,
+                                            timeout=timeout)
+            self.path = path
+
+        def connect(self):
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            sock.connect(self.path)
+            self.sock = sock
+except ImportError:
+    # Python 3
+    import http.client
+
+    class UnixHTTPConnection(http.client.HTTPConnection):
+        def __init__(self, path, host='localhost', port=None,
+                     timeout=None):
+            http.client.HTTPConnection.__init__(self, host, port=port,
+                                                timeout=timeout)
+            self.path = path
+
+        def connect(self):
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            sock.connect(self.path)
+            self.sock = sock
```

Comment:
Do you think it would be possible te declare this class only once using a signature like `__init__(self, path, *args, **kwargs)` ?